android/jni/com/mapswithme/maps/MwmApplication.cpp

```diff
@@ -13,18 +13,20 @@
 extern "C"
 {
   JNIEXPORT void JNICALL
-  Java_com_mapswithme_maps_MwmApplication_nativeInit(
-      JNIEnv * env, jobject thiz,
-      jstring apkPath, jstring storagePath, jstring tmpPath, jstring obbGooglePath,
-      jstring flavorName, jstring buildType, jboolean isYota, jboolean isTablet)
+  Java_com_mapswithme_maps_MwmApplication_nativeInit(JNIEnv * env, jobject thiz,
+                                                     jstring apkPath, jstring settingsPath, jstring tmpPath, jstring obbGooglePath,
+                                                     jstring flavorName, jstring buildType, jboolean isTablet)
   {
-    android::Platform::Instance().Initialize(
-        env, apkPath, storagePath, tmpPath, obbGooglePath, flavorName, buildType, isYota, isTablet);
+    android::Platform::Instance().Initialize(env, apkPath, settingsPath, tmpPath, obbGooglePath, flavorName, buildType, isTablet);
+  }
 
-    LOG(LDEBUG, ("Creating android::Framework instance ..."));
+  JNIEXPORT void JNICALL
+  Java_com_mapswithme_maps_MwmApplication_nativeStart(JNIEnv * env, jobject thiz, jstring mapsPath)
+  {
+    ASSERT(!g_framework, ("nativeStart() must NOT be called twice!"));
 
-    if (!g_framework)
-      g_framework = new android::Framework();
+    android::Platform::Instance().SetStoragePath(jni::ToNativeString(env, mapsPath));
+    g_framework.reset(new android::Framework());
```

Comment:
If it is always called once, then ```ASSERT(g_framework == false, ());``` is needed.
If it can be called more than once, then ```g_framework.reset(nullptr); g_framework.reset(new android::Framework());``` is needed.